lib/improver/utilities/spatial.py

```diff
@@ -0,0 +1,168 @@
+        diff_cube = Cube(diff_along_axis, **metadata_dict)
+
+        for coord in cube.dim_coords:
+            dims = cube.coord_dims(coord)
+            if coord.name() in [coord_name]:
+                coord = DimCoord(
+                    mean_points, standard_name=coord.standard_name,
+                    units=coord.units)
+            diff_cube.add_dim_coord(coord.copy(), dims)
+        for coord in cube.aux_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+        for coord in cube.derived_coords:
+            dims = cube.coord_dims(coord)
+            diff_cube.add_aux_coord(coord.copy(), dims)
+
+        # Add metadata to indicate that a difference has been calculated.
+        # TODO: update this metadata when proper conventions have been
+        #       agreed upon.
+        cell_method = CellMethod("difference", coords=[coord_name],
+                                 intervals='1 grid length')
+        diff_cube.add_cell_method(cell_method)
+        diff_cube.attributes["form_of_difference"] = (
+            "forward_difference")
+        diff_cube.rename('difference_of_' + cube.name())
+        print diff_cube
+        return diff_cube
+
+    def calculate_difference(self, cube, coord_axis):
+        """
+        Calculate the difference along the axis specified by the
+        coordinate.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the differences will be calculated.
+        coord_axis : String
+            Short-hand reference for the x or y coordinate, as allowed by
+            iris.util.guess_coord_axis.
+
+        Returns
+        -------
+        diff_cube : Iris.cube.Cube
+            Cube after the differences have been calculated along the
+            specified axis.
+        """
+        coord_name = cube.coord(axis=coord_axis).name()
+        diff_axis = cube.coord_dims(coord_name)[0]
+        diff_along_axis = np.diff(cube.data, axis=diff_axis)
+        diff_cube = self.create_difference_cube(
+            cube, coord_name, diff_along_axis)
+        return diff_cube
+
+    def process(self, cube):
+        """
+        Calculate the difference along the x and y axes and return
+        the result in separate cubes. The difference along each axis is
+         calculated using numpy.diff.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the differences will be calculated.
+
+        Returns
+        -------
+        diff_along_y_cube : Iris.cube.Cube
+            Cube after the differences have been calculated along the
+            y axis.
+        diff_along_x_cube : Iris.cube.Cube
+            Cube after the differences have been calculated along the
+            x axis.
+
+        Raises
+        ------
+        InvalidCubeError if the input cube has too many dimensions.
+        """
+        if cube.ndim != 2:
```

Comment:
If 3D or 4D cubes can be supported without looping over slices (which I think this code can do) then I think this condition should be removed. It will then need a new unit test. Sorry!